src/master/master_heartbeat.cpp

```diff
@@ -1334,6 +1340,16 @@ hb_cluster_calc_score (void)
 	}
     }
 
+  cubhb::node_entry *new_master = hb_Cluster->master;
+  if (old_master != NULL && new_master != NULL && old_master->get_hostname () != new_master->get_hostname ())
```

Comment:
Theoretically hb_cluster->master is never NULL, it is more a safety check.

Reply:
then please add debug safe-guards that the pointers are never null. otherwise it may be completely missed.

Reply:
Actually only new_master cannot be NULL (old_master can be NULL at boot time).
Added: `assert (new_master != NULL);` and trigger send hostname if:
- old_master is NULL and new_master is not NULL (this will happen at boot time), therefore the job was removed from `hb_resource_job_initialize` function because it does same thing
- both new and old master are not NULL and the hostname was changes